Azure/sonic-buildimage
# [cfggen] Extend Template Argument to Support Batch Mode

Calls to cfggen take considerable time. With batch mode, we will have the ability 
to reduce number of calls from service to one call only.

Example of the batch mode command:
`sonic-cfggen -t template-1.j2 -t template-2.j2,config-db -t template-3.j2,config-db -t template-4.j2,file1 -t template-5.j2,file2 --write-to-db`. 

`template-1.j2` will be rendered to stdout since it is missing the dest part. stdout is default
`config-db` is a special keyword that will inject the rendered template into internal data structure. The internal data structure gets written to redis-db with `--write-to-db` switch. In the case the user would like to write to a file named config-db, it could be given as `/config-db` or `./config-db`

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

**- Why I did it**
In order to reduce number of calls being made to sonic-cfggen during boot process

**- How I did it**
Added new mode (batch) to cfggen where multiple templates could be rendered and written to disk at once

**- How to verify it**
1. Provided test case to verify the functionality
2. Incorporate this new batch into interfaces-config.sh and I could see gain of ~3 sec
```
admin@str-s6000-acs-14:~$ time sudo ./interfaces-config-old.sh

real	0m4.826s
user	0m3.854s
sys	0m0.769s
admin@str-s6000-acs-14:~$ time sudo ./interfaces-config-new.sh

real	0m1.835s
user	0m1.444s
sys	0m0.316s
admin@str-s6000-acs-14:~$ time sudo ./interfaces-config-old.sh

real	0m4.743s
user	0m3.747s
sys	0m0.708s
admin@str-s6000-acs-14:~$ time sudo ./interfaces-config-new.sh

real	0m1.636s
user	0m1.297s
sys	0m0.268s
```
**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
